plugin/src/main/java/org/apache/jackrabbit/filevault/maven/packaging/VaultMojo.java

```diff
@@ -534,6 +489,94 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     }
 
     /**
+     * Computes the package filters.
+     *
+     * Requirements:
+     * - backward compatibility: if a filter.xml is copied to vault-work with the resource plugin, then it should still "work" correctly.
+     * - if there are any comments in the original filter source, they should not get lost, if possible
+     * - if there are filters specified in the pom and in a filter source, they should get merged.
+     * - if the prefix property is set, it should be used if no filter is set.
+     * - if both, a inline filter and a implicit filter is present, the build fails.
+     * - re-run the package goal w/o cleaning the target first must work
+     *
+     * @throws IOException if an I/O error occurs
+     * @throws MojoExecutionException if the build fails
+     */
+    private void computePackageFilters(File vaultMetaDir) throws IOException, MojoExecutionException {
+        // backward compatibility: if implicit filter exists, use it. but check for conflicts
+        File filterFile = new File(vaultMetaDir, "filter.xml");
+        if (filterFile.exists() && filterFile.lastModified() != 0) {
+            // if both, a inline filter and a implicit filter is present, the build fails.
+            if (!filters.getFilterSets().isEmpty()) {
+                getLog().error("Refuse to merge inline filters and non-sourced filter.xml. If this is intended, specify the filter.xml via the 'filterSource' property.");
+                throw new MojoExecutionException("conflicting filters.");
+            }
+            // load filters for further processing
+            filters.load(filterFile);
+
+            getLog().warn("The project is using a filter.xml provided via the resource plugin.");
+            getLog().warn("This is deprecated and might no longer be supported in future versions.");
+            getLog().warn("Use the 'filterSource' property to specify the filter or use inline filters.");
+            return;
+        }
+
+        // if last modified of vault-work/META-INF/vault/filter.xml == 0 -> delete it
+        if (filterFile.exists() && filterFile.lastModified() == 0) {
+            if (!filterFile.delete()) {
```

Comment:
Rather use Files#delete() (https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#delete(java.nio.file.Path)) to get a helpful exception in case deletion fails. We require Java7 anyhow.